code/modules/admin/verbs/request_music.dm

```diff
@@ -6,24 +6,28 @@
 		to_chat(usr, span_danger("Speech is currently admin-disabled."), confidential = TRUE)
 		return
 
-	var/msg = tgui_input_text(usr.client, "Please Input a URL", "Music Request", "")
-	if(!msg)
+	var/request_url = tgui_input_text(usr.client, "Please Input a URL", "Music Request", "")
+	if(!request_url)
 		return
-	log_prayer("[src.key]/([src.name]): [msg]")
+
+	//regex filter
+	if(!findtext(request_url, "youtube.com/watch?v=") && !findtext(request_url, "youtu.be/") && !findtext(request_url, "soundcloud.com/"))
```

Comment:
lemme use bandcamp :-)

Reply:
does youtube-dl even support bandcamp?

Reply:
oh it does, alright, thats a game changer i never knew that

Reply:
given youtube-dl supports every single porn site i've ever heard of, i think its a fair chance it supports bandcamp as well as nearly every single music streaming site that is well known.

Like, it fucking supports ripping songs off of myspace: https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/extractor/myspace.py

You shouldn't even need to ask if it supports it.